### PR TITLE
pallet migration to FRAME v2 & refactor weights

### LINKED ISSUE #102: Refactor Weights
## Describe the enhancement
After #88 refactor the pallet's weights and make sure they make sense

### PR DESCRIPTION
**Pull request by shamb0**
pallet migration to FRAME v2 & refactor weights

fixes #102

### CODE REVIEW

File: pallets/amendments/src/lib.rs [hunk 1]
```diff
@@ -21,88 +21,79 @@
...
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
         /// Schedule `amendment` to be executed after the configured time, unless vetoed by `VetoOrigin`
-        #[weight = 100_000_000]
-        fn propose(origin, amendment: Box<T::Amendment>) -> DispatchResult {
+        #[pallet::weight(T::WeightInfo::propose(*_length_bound))]
+        pub fn propose(
+            origin: OriginFor<T>,
+            amendment: Box<T::Amendment>,
+            _length_bound: u32,
```
Comment (ETeissonniere) [inline pallets/amendments/src/lib.rs:88]: I think we should check that `length_bound >= size_of_amendment` and throw an error if this is not the case

**Review by ETeissonniere — CHANGES_REQUESTED** [review]
Thanks! I only have one tiny comment :)

File: pallets/amendments/src/lib.rs (hunk 1 above)
Comment (shamb0) [reply pallets/amendments/src/lib.rs:88]: Super Fast, Thanks for quick comment. Sure will rework on it :+1:

Comment (shamb0) [reply pallets/amendments/src/lib.rs]: Hello Eliott,

Completed refactoring weights annotation for `pub fn propose()`, as discussed removed the extra argument of ` _length_bound: u32`.

**Review by ETeissonniere — APPROVED** [review]
Very nice, thanks!

**Pull request merged**